R/draws-index.R

```diff
@@ -141,6 +141,43 @@ remove_reserved_variable_names <- function(variables, reserved) {
   x
 }
 
+#' Set variable names in `draws` objects
+#'
+#' Set variable names for all variables in a [`draws`] object. Useful
+#' when using pipe operators.
+#'
+#' @param x (draws) A [`draws`] object.
+#' @param variables (character) new variable names.
+#' @template args-methods-dots
+#'
+#' @return Returns a [`draws`] object of the same format as `x`, with
+#'   variables named as specified.
+#'
+#' @seealso [`variables`]
+#' @examples
+#' x <- as_draws(matrix(rnorm(100), ncol = 2))
+#' variables(x)
+#'
+#' set_variables(x, c("theta[1]", "theta[2]"))
+#' variables(x)
+#'
+#' # this is equivalent to
+#' variables(x) <- c("theta[1]", "theta[2]")
+#' variables(x)
+#'
+#' @export
+set_variables <- function(x, variables, ...) {
+  UseMethod("set_variables")
+}
+
+#' @rdname set_variables
+#' @export
+set_variables.draws <- function(x, variables, ...) {
```

Comment:
Why make `set_variables` generic? If it is a non-generic function that just forwards to `variables<-` then implementers of other classes need only override the generic there.

Reply:
I didn't think too much about this either way, as I just used `rename_variables` as a template, which is a generic with only a `rename_variables.draws` method defined. If it's better to change it to a non-generic function, I can do that

Reply:
I defer to @paul-buerkner, but IMO it would be simpler :).

Reply:
non-generic sounds good to me too

Reply:
I changed it to non-generic now